pygmt/src/pygmtlogo.py

```diff
@@ -0,0 +1,269 @@
+        case "dark":
+            color_bg = color_dark
+            color_py = color_yellow
+            color_gmt = color_light
+
+    # Define shape
+    match shape:
+        case "circle":
+            symbol = "c"  # circle
+            diameter = 7.5
+            diameter_add = 0.5
+        case "hexagon":
+            symbol = "h"  # hexagon
+            diameter = 8.6
+            diameter_add = 0.6
+
+    # Define wordmark
+    font = "AvantGarde-Book"
+    match wordmark:
+        case "vertical":
+            args_text_wm = {"x": 0, "y": -5, "justify": "CT", "font": f"2.5c,{font}"}
+        case True | "horizontal":
+            args_text_wm = {"x": 6, "y": 0, "justify": "LM", "font": f"8c,{font}"}
+
+    fig = pygmt.Figure()
+    fig.basemap(
+        region=region, 
+        projection=projection, 
+        perspective=perspective, 
+        frame="+n",  # Change it to `frame="afg"` for debugging.
+    )
+
+    # blue circle / hexagon for Earth
+    fig.plot(
+        x=0,
+        y=0,
+        style=f"{symbol}{diameter}c",
+        pen=f"0.5c,{color_blue}",
+        fill=color_bg,
+        perspective=True,
+        no_clip=True,  # needed for corners of hexagon shape
+    )
+
+    # Compass
+    x1, x2 = r1 * 0.7071, r2 * 0.7071  # sqrt(2)/2 = 0.7071
+    lines_compass = [
+        ([-r0, -r2], [0, 0]),  # horizontal lines
+        ([-r4, 0], [0, 0]),
+        ([r2, r0], [0, 0]),
+        ([0, 0], [-r0, r0]),  # vertical line
+        ([-x1, -x2], [x1, x2]),  # upper left
+        ([-x1, -x2], [-x1, -x2]),  # lower left
+        ([x1, x2], [x1, x2]),  # upper right
+        ([x1, x2], [-x1, -x2]),  # lower right
+    ]
+    for x, y in lines_compass:
+        fig.plot(x=x, y=y, pen=f"5p,{color_yellow}", perspective=True)
+
+    # Letter G
+    angles = np.deg2rad(np.arange(90, 361, 1.0))
+    x = np.concatenate(
+        [np.cos(angles) * r3, [r3, 0, 0, r4], np.cos(np.flip(angles)) * r4]
+    )
+    y = np.concatenate(
+        [
+            np.sin(angles) * r3,
+            [(r3 - r4) / 2.0, (r3 - r4) / 2.0, -(r3 - r4) / 2.0, -(r3 - r4) / 2.0],
+            np.sin(np.flip(angles)) * r4,
+        ]
+    )
+    fig.plot(x=x, y=y, fill=color_red, perspective=True)
+
+    # upper vertical red line
+    # space between red line and blue circle / hexagon
+    fig.plot(x=[0, 0], y=[r0, r2], pen=f"18p,{color_bg}", perspective=True)
+    # red line
+    fig.plot(x=[0, 0], y=[r0, r2], pen=f"12p,{color_red}", perspective=True)
+
+    # letter M
+    # space between letter M and yellow line at the right side
+    # fig.plot(x=[1.6, 1.6], y=[1.5, 1.775], pen=f"10p,{color_bg}")
+    fig.plot(x=[1.6, 1.6], y=[1.5, 2.0], pen=f"10p,{color_bg}", perspective=True)
+
+    # polygon with small distance to horizontal line of letter G
+    # starting point: lower right corner of the left vertical line of letter M
+    # direction: clockwise
+    m_x1 = 0.33 - 0.33 / 2 - 0.06
+    m_x2 = 1.54 + 0.33 / 2 - 0.06  # outer radius of letter G
```

Comment:
This is actually equal to `r3`:
```suggestion
    m_x2 = r3
```